samples: Adapt DTM to Sysbuild (#15143)
nrfconnect/sdk-nrf: samples/bluetooth/direct_test_mode/README.rst

* A sample that provides an external interface to the library.

You can find the source code of both parts in :file:`samples/bluetooth/direct_test_mode/src`.

> If you use experimental HCI interface on |nRF5340DKnoref| you can fing additional source code in :file:`samples/bluetooth/direct_test_mode/remote_hci/src` and :file:`samples/bluetooth/direct_test_mode/rpc`.

Reviewer: ```suggestion
If you use the experimental HCI interface on the |nRF5340DKnoref| you can find additional source code in :file:`samples/bluetooth/direct_test_mode/remote_hci/src` and :file:`samples/bluetooth/direct_test_mode/rpc`.
```

> If you use the experimental HCI interface on the |nRF5340DKnoref| you can find additional source code in :file:`samples/bluetooth/direct_test_mode/remote_hci/src` and :file:`samples/bluetooth/direct_test_mode/rpc`.

Reviewer: ```suggestion
If you use the experimental HCI interface on the |nRF5340DKnoref|, you can find additional source code in :file:`samples/bluetooth/direct_test_mode/remote_hci/src` and :file:`samples/bluetooth/direct_test_mode/rpc`.
```

The default selection of UART pins is defined in :file:`zephyr/boards/arm/board_name/board_name.dts`.
You can change the defaults using the symbols ``tx-pin`` and ``rx-pin`` in the DTS overlay file of the child image at the project level.

> The configuraion files for the :ref:`nrf5340_remote_shell` subimage are located in the :file:`sysbuild/remote_shell` directory.

Reviewer: ```suggestion
The configuration files for the :ref:`nrf5340_remote_shell` subimage are located in the :file:`sysbuild/remote_shell` directory.
```

.. code-block:: console

   west build samples/bluetooth/direct_test_mode -b board_name --sysbuild -- -DSB_CONFIG_DTM_TRANSPORT_HCI=y

> On the |nRF5340DKnoref| the sample with HCI interface can also be built with the `remote_hci` image using the same command.

Reviewer: ```suggestion
On the |nRF5340DKnoref|, you can build the sample with HCI interface with the ``remote_hci`` image using the same command.
```
